server/src/tests/helpers.rs

```diff
@@ -102,21 +100,21 @@ pub(crate) async fn server_with_handles() -> (SocketAddr, ServerHandle) {
 	module.register_method("notif", |_, _| "").unwrap();
 	module
 		.register_method("should_err", |_, ctx| {
-			ctx.err().map_err(CallError::Failed)?;
+			ctx.err()?;
 			RpcResult::Ok("err")
 		})
 		.unwrap();
 
 	module
 		.register_method("should_ok", |_, ctx| {
-			ctx.ok().map_err(CallError::Failed)?;
+			ctx.ok()?;
 			RpcResult::Ok("ok")
 		})
 		.unwrap();
 	module
-		.register_async_method("should_ok_async", |_p, ctx| async move {
-			ctx.ok().map_err(CallError::Failed)?;
-			Result::<_, Error>::Ok("ok")
+		.register_async_method::<Result<&str, MyAppError>, _, _>("should_ok_async", |_p, ctx| async move {
+			ctx.ok()?;
+			Ok("ok")
```

Comment:
Nit: maybe consistify these generic ones with what you did below with §Result::<_, MyAppError>::Ok(..)`